flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -321,6 +323,14 @@ public void setZookeeperNamespace(String zookeeperNamespace) {
 		this.zookeeperNamespace = zookeeperNamespace;
 	}
 
+	public String getNodeLabel() {
+		return nodeLabel;
+	}
+
+	public void setNodeLabel(String nodeLabel) {
+		this.nodeLabel = nodeLabel;
+	}
```

Comment:
We should get rid of this setter pattern. All values should be set at creation time of the `AbstractYarnClusterDescriptor`.

Reply:
@tillrohrmann the setter for `nodelabel` I followed the example of `setZookeeperNamespace` and `setQueue`, it seems that the `AbstractYarnClusterDescriptor` constructor can not set this field if we do not change it.

Reply:
Yes, at some point we should change it in order to make it better maintainable. This will of course not be in the scope of this PR.